### PR TITLE
Fat 31 - tests for mod-email

### PR DESCRIPTION
**Pull request by viktor-draban**
https://issues.folio.org/browse/FAT-31

### CODE REVIEW

**Review by Mykyta-Varenyk — APPROVED** [review]

File: mod-email/src/main/resources/domain/mod-email/features/delayedTasks.feature [hunk 1]
```diff
@@ -5,14 +5,22 @@ Feature: Delayed tasks
...
 
-  @Undefined
   Scenario: Delete should return 204 on success
-    * print 'undefined'
+    Given path 'delayedTask/expiredMessages'
+    When method DELETE
+    Then status 204
 
-  @Undefined
   Scenario: Delete should return 400 if bad request
```
Comment (roman-barannyk) [inline mod-email/src/main/resources/domain/mod-email/features/delayedTasks.feature:13]: if invalid token?

File: mod-email/src/main/resources/domain/mod-email/features/email.feature [hunk 2]
```diff
@@ -9,27 +9,44 @@ Feature: Email
...
     Then status 200
+    And match response == { emailEntity: #present, totalRecords: #present }
 
-  @Undefined
-  Scenario: Get email should return 401 if unauthorized error
-    * print 'undefined'
-
-  @Undefined
-  Scenario: Get email should return 404 if email is not found
-    * print 'undefined'
```
Comment (roman-barannyk) [inline mod-email/src/main/resources/domain/mod-email/features/email.feature:19]: why have these cases been removed?

File: mod-email/src/main/resources/domain/mod-email/features/email.feature [hunk 3]
```diff
@@ -9,27 +9,44 @@ Feature: Email
...
+    * def expectedErrMsg = 'The \'mod-config\' module doesn\'t have a minimum config for SMTP server, the min config is: [EMAIL_SMTP_PORT, EMAIL_PASSWORD, EMAIL_SMTP_HOST, EMAIL_USERNAME]'
+
+    Given path 'email'
+    And request requestEntity
+    When method POST
+    Then status 200
+    And match response == expectedErrMsg
 
-  @Undefined
   Scenario: Post email should return 400 if bad request
```
Comment (roman-barannyk) [inline mod-email/src/main/resources/domain/mod-email/features/email.feature:32]: if invalid token?

**Review by roman-barannyk — APPROVED** [review]

File: mod-email/src/main/resources/domain/mod-email/features/email.feature (hunk 2 above)
Comment (viktor-draban) [reply mod-email/src/main/resources/domain/mod-email/features/email.feature:19]: even if the endpoint doesn't find any email it returns status code 200 with empty result, so this test-case is impassible with current implementation

File: mod-email/src/main/resources/domain/mod-email/features/delayedTasks.feature (hunk 1 above)
Comment (viktor-draban) [reply mod-email/src/main/resources/domain/mod-email/features/delayedTasks.feature:13]: I didn't find another way to get 400

File: mod-email/src/main/resources/domain/mod-email/features/email.feature (hunk 3 above)
Comment (viktor-draban) [reply mod-email/src/main/resources/domain/mod-email/features/email.feature:32]: I didn't find another way to get 400

File: mod-email/src/main/resources/domain/mod-email/features/delayedTasks.feature (hunk 1 above)
Comment (roman-barannyk) [reply mod-email/src/main/resources/domain/mod-email/features/delayedTasks.feature:13]: then I would suggest renaming the scenario

Comment (viktor-draban) [reply mod-email/src/main/resources/domain/mod-email/features/delayedTasks.feature:13]: In which name do you suggest rename this test?

**Pull request merged**